sdk/src/azure/core/az_hex_private.h

```diff
@@ -19,6 +19,7 @@ enum
  */
 AZ_NODISCARD AZ_INLINE uint8_t _az_number_to_upper_hex(uint8_t number)
 {
+  // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
```

Comment:
add an enum as well?  
just like for the other fixes

Reply:
I intentionally avoided adding named symbols for things that are one-off, or didn't have a useful name outside of the local instance. If desired, we can add them later.

Got a suggestion for the name here? The 10 here is essentially one more than the last non-alphabetic digit of the hex values (0-9, A-F).
I couldn't come up with something useful, and this is not being shared anywhere else, so left it as is.

Reply:
maybe just add this explanation about what 10 means as a comment and that's fine